packages/@o3r/rules-engine/src/components/rules-engine/facts-snapshot/facts-snapshot.component.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  input,
+  signal,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  FormsModule,
+  ReactiveFormsModule,
+} from '@angular/forms';
+import type {
+  Facts,
+} from '../../../engine';
+import {
+  O3rJsonOrStringPipe,
+} from '../shared/index';
+
+@Component({
+  selector: 'o3r-facts-snapshot',
+  styleUrls: ['./facts-snapshot.style.scss'],
+  templateUrl: './facts-snapshot.template.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    O3rJsonOrStringPipe,
+    FormsModule,
+    ReactiveFormsModule
+  ],
+  encapsulation: ViewEncapsulation.None
+})
+export class FactsSnapshotComponent {
+  /**
+   * Full list of available facts with their current value
+   */
+  public facts = input<{ factName: string; value: Facts }[]>([]);
```

Comment:
Shouldn't it be required ?
```suggestion
  public facts = input.required<{ factName: string; value: Facts }[]>();
```